src/MySqlConnector/MySqlClient/MySqlErrorCode.cs

```diff
@@ -0,0 +1,18 @@
+namespace MySql.Data.MySqlClient
+{
+	/// <summary>
+	/// MySQL Server error codes. Taken from <a href="https://dev.mysql.com/doc/refman/5.7/en/error-messages-server.html">Server Error Codes and Messages</a>.
+	/// </summary>
+	public enum MySqlErrorCode
```

Comment:
This is for eventual API compatibility: #232

(We could also code-gen the class you suggest although I'm not sure the duplication would be worth it.)